task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -422,16 +422,15 @@ def run(ti: RuntimeTaskInstance, log: Logger):
         # updated already be another UI API. So, these exceptions should ideally never be thrown.
         # If these are thrown, we should mark the TI state as failed.
         msg = TaskState(
-            state=TerminalTIState.FAILED,
+            state=TerminalTIState.FAIL_WITHOUT_RETRY,
```

Comment:
I think this one should still be FAILED?

Reply:
Now I think about it: I'm not sure this exception can actually ever be thrown, it looks like before that TI.heartbeat would notice the state and throw this exception "in" to the task? Not sure, lets look at that in a future PR

Reply:
Alright, will investigate in a follow up. For what is worth, it should never be thrown actually. This is just a safety hatch in case it is